sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -855,7 +855,9 @@ static void _receive(gnrc_pktsnip_t *pkt)
         ipv6 = gnrc_pktbuf_mark(pkt, sizeof(ipv6_hdr_t), GNRC_NETTYPE_IPV6);
 
         first_ext = pkt;
+#ifndef MODULE_GNRC_SIXLOWPAN 
         pkt->type = GNRC_NETTYPE_UNDEF; /* snip is no longer IPv6 */
```

Comment:
This line makes sense only for 6LoWPAN packets. See https://github.com/RIOT-OS/RIOT/pull/5596#issuecomment-234821556. So `#ifndef MODULE_GNRC_SIXLOWPAN` doesn't make sense.
If we doesn't set `pkt->type = GNRC_NETTYPE_UNDEF` when the 6LoWPAN module is enabled, other modules (e.g. gnrc_icmpv6, gnrc_udp) will be confused because a snip without IPv6 header is marked as a IPv6 snip. Recall https://github.com/RIOT-OS/RIOT/pull/5596#r69503889.